Update sra chart to use unified docker image both for ssh and web bastions

avishay28 opened the pull request:
We should decide what to do with the applications version.
I guess we want to use the same version for both apps (`sshVersion`,`ztbVersion`)

In that case we need to update the annotations in Chat.yaml to use only one annotation, we will discuess it..
Currently it's:
```
annotations:
  ztbVersion: v1.0.0
  sshVersion: 0.16.1 // need also to be v1.0.0?
  ztpVersion: 1.4.2
```
And it being concatenated: `appVersion: v1.0.0_0.16.1_1.4.2`

This can be merged only after @omriezra finish the pipline and the updated image is pushed to `akeyless/zero-trust-bastion` as latest.

rani-sharim commented on charts/akeyless-secure-remote-access/Chart.yaml:
```diff
@@ -14,7 +14,7 @@ type: application
 # to the chart and its templates, including the app version.
 
-version: 0.28.5
+version: 0.28.6
```
This is at least a minor version imo

avishay28 replied:
change to `0.29.5`

renanaAkeyless replied:
no please change to 0.29.0

rani-sharim commented on charts/akeyless-secure-remote-access/values.yaml:
```diff
@@ -255,7 +258,10 @@ ztpConfig:
-  env: []
+  env:
+    # This env variable instruct docker to start only web-bastion processes
+    - name: "BASTION"
```
Do we need this for the `portal` as well?

avishay28 replied:
no, only `ztb` and `sshProxy`, i udated the code

rani-sharim commented on charts/akeyless-secure-remote-access/values.yaml:
```diff
@@ -367,15 +373,19 @@ sshConfig:
 
   image:
-    repository: akeyless/ssh-proxy
+    repository: akeyless/zero-trust-portal
```
Is this the correct repo?
Shouldn't the repo for `zero trust bastion` be the same?

avishay28 replied:
sshProxy and web bastion should have the same repo - `akeyless/zero-trust-bastion`, the portal stays the same (udpdated the code)

renanaAkeyless commented on charts/akeyless-secure-remote-access/values.yaml:
```diff
@@ -99,7 +99,10 @@ ztbConfig:
-  env: []
+  env:
+    # This env variable instruct docker to start only web-bastion processes
+    - name: "BASTION"
```
why not BASTION_TYPE ?
BASTION is a bit confusing name

renanaAkeyless replied:
dont add this to env part.. put it in a separate section

avishay28 replied:
done

renanaAkeyless commented on charts/akeyless-secure-remote-access/values.yaml:
```diff
@@ -367,15 +370,19 @@ sshConfig:
   image:
-    repository: akeyless/ssh-proxy
+    repository: akeyless/zero-trust-bastion
     pullPolicy: Always
```
pullPolicy: Always? why not ifnotpresent ???

avishay28 replied:
idk, it was like this before, didn't change it..

avishay28 replied:
@niv-ez @omriezra there is a reason for it? or can i change it to `ifnotpresent`?

OriMankali replied:
Agreed

renanaAkeyless commented on charts/akeyless-secure-remote-access/values.yaml:
```diff
@@ -367,15 +370,19 @@ sshConfig:
-  env: []
+  env:
+    # This env variable instruct docker to start only ssh-proxy-bastion processes
+    - name: "BASTION"
```
same on the name

renanaAkeyless replied:
dont add this to env part.. put it in a separate section

avishay28 replied:
done

renanaAkeyless reviewed (changes requested):
please fix my comments

idanmantin reviewed (commented):
withdraw approval

avishay28 commented on charts/akeyless-secure-remote-access/Chart.yaml:
```diff
@@ -14,14 +14,14 @@ type: application
-  sshVersion: 0.16.1
+  ztbVersion: v2.0.0
+  sshVersion: v2.0.0
   ztpVersion: 1.4.2
```
@OriMankali @renanaAkeyless @niv-ez please confirm the app verstion.
What the new version should be?

niv-ez replied:
I believe that ZTB and SSH can be the same version but ZTP must stay independent, If a change you make on ZTB will also be applied to SSH then it's OK but as long as their releases are not match then leave it as is for now

OriMankali replied:
I agree, you can also remove the 'v' prefix, it was an historic misalignment (just check there are no functional implications - e.g. validation of version, etc.)
We can just use ztbVersion and remove the ssh one

OriMankali replied:
Please make sure not to break bc, the version should be propagated to the same location the application was expecting

niv-ez replied:
@omriezra Please check the caller git repositories to make sure they will bump the same version and verify that the Helm chart release scrip in this repo knows how to handle it

avishay28 replied:
@OriMankali there are multiple places where we expect the version to start with "v" like here
https://github.com/akeylesslabs/zero-trust-bastion/blob/c4cf5ad327501545148c6451a5f76c568c57a252/.github/scripts/release_latest_version.sh#L15

@omriezra can we modify the pipline to not depend on this requirment?

avishay28 replied:
currently i leave it with both have the same version `v2.0.0`, @omriezra  please update me if we can remove the 'v' prefix.

OriMankali commented on charts/akeyless-secure-remote-access/README.md:
```diff
@@ -11,7 +11,7 @@ with signed certificate authentication, together with session recording.
 ## Introduction
 
-This chart bootstraps a Akeyless Zero Trust Bastion deployment,  Akeyless SSH bastion statefulset and a Akeyless Zero Trust Portal deployment on Kubernetes 
+This chart bootstraps a Akeyless Zero Trust Bastion deployment,  Akeyless SSH bastion statefulset and a Akeyless Zero Trust Portal deployment on Kubernetes
```
It looks like we should not mention SSH bastion anymore, and just have ZTB (now statefulset) and ZTP, no?

avishay28 replied:
Not sure I undersatsnd, from what I understand from what we speak with @niv-ez, we want at first drop to make minimal changes to the chart- still have both services (ssh and ztb), just change the docker image (becauase now they use the same image), and in each service pass the relevant mode (I called this env variable `BASTION_TYPE`)
so ssh service start with `BASTION_TYPE=ssh-proxy` and web service start with `BASTION_TYPE=web`, but still these two services present in the chart.

OriMankali reviewed (commented):
Is there no flag for operating in ssh mode? is it already serving both apps?

avishay28 commented:
@OriMankali  There is a flag to operate in SSH mode, the env variable `BASTION_TYPE=ssh-proxy` that i added in `charts/akeyless-secure-remote-access/templates/statefulSet.yaml` for ssh service,
and `BASTION_TYPE=web` I added here `charts/akeyless-secure-remote-access/templates/deployment.yaml`, for ztb service.
At the begning I added it in the values.yaml, but renana asked me to make it hardcoded because users shoulnd't be able to change this..

omriezra commented on charts/akeyless-secure-remote-access/Chart.yaml:
```diff
@@ -14,14 +14,13 @@ type: application
 
 
-appVersion: v1.0.0_0.16.1_1.4.2
+appVersion: 2.0.0_1.4.2
```
```suggestion
appVersion: 1.0.0_1.4.2
```

version will be bump to 2 after release

omriezra commented on charts/akeyless-secure-remote-access/Chart.yaml:
```diff
@@ -14,14 +14,13 @@ type: application
 annotations:
-  ztbVersion: v1.0.0
-  sshVersion: 0.16.1
+  ztbVersion: 2.0.0
```
```suggestion
  ztbVersion: 1.0.0
```
